fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -366,12 +441,22 @@ public Table getView(String dbName, String viewName) {
         }
     }
 
-    @Override
-    public TableVersionRange getTableVersionRange(Table table) {
-        IcebergTable icebergTable = (IcebergTable) table;
-        Optional<Long> snapshotId = Optional.ofNullable(icebergTable.getNativeTable().currentSnapshot())
-                .map(Snapshot::snapshotId);
-        return TableVersionRange.withEnd(snapshotId);
+    public TableVersionRange getTableVersionRange(Table table,
+                                                  Optional<ConnectorTableVersion> startVersion,
+                                                  Optional<ConnectorTableVersion> endVersion) {
+        if (startVersion.isPresent()) {
```

Comment:
I think this check should in this connector Metadata. I don't known what will happen when start version is present for other catalog

Reply:
if table support incremental query based on between .. and with snapshotId, we can use this startVersion.